controllers/provider-openstack/charts/internal/openstack-infra/templates/_main.tf

```diff
@@ -0,0 +1,129 @@
+}
+{{- end}}
+
+resource "openstack_networking_network_v2" "cluster" {
+  name           = "{{ required "clusterName is required" .Values.clusterName }}"
+  admin_state_up = "true"
+}
+
+resource "openstack_networking_subnet_v2" "cluster" {
+  name            = "{{ required "clusterName is required" .Values.clusterName }}"
+  cidr            = "{{ required "networks.worker is required" .Values.networks.worker }}"
+  network_id      = "${openstack_networking_network_v2.cluster.id}"
+  ip_version      = 4
+  {{- if .Values.dnsServers }}
+  dns_nameservers = [{{- include "openstack-infra.dnsServers" . | trimSuffix ", " }}]
+  {{- else }}
+  dns_nameservers = []
+  {{- end }}
+}
+
+resource "openstack_networking_router_interface_v2" "router_nodes" {
+  router_id = "{{ required "router.id is required" $.Values.router.id }}"
+  subnet_id = "${openstack_networking_subnet_v2.cluster.id}"
+}
+
+resource "openstack_networking_secgroup_v2" "cluster" {
+  name                 = "{{ required "clusterName is required" .Values.clusterName }}"
+  description          = "Cluster Nodes"
+  delete_default_rules = true
+}
+
+resource "openstack_networking_secgroup_rule_v2" "cluster_self" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  security_group_id = "${openstack_networking_secgroup_v2.cluster.id}"
+  remote_group_id   = "${openstack_networking_secgroup_v2.cluster.id}"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "cluster_egress" {
+  direction         = "egress"
+  ethertype         = "IPv4"
+  security_group_id = "${openstack_networking_secgroup_v2.cluster.id}"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "cluster_tcp_all" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  port_range_min    = 1
+  port_range_max    = 65535
+  remote_ip_prefix  = "0.0.0.0/0"
+  security_group_id = "${openstack_networking_secgroup_v2.cluster.id}"
+}
+
+//=====================================================================
+//= SSH Key for Nodes (Bastion and Worker)
+//=====================================================================
+
+resource "openstack_compute_keypair_v2" "ssh_key" {
+  name       = "{{ required "clusterName is required" .Values.clusterName }}"
+  public_key = "{{ required "sshPublicKey is required" .Values.sshPublicKey }}"
+}
+
+// We have introduced new output variables. However, they are not applied for
+// existing clusters as Terraform won't detect a diff when we run `terraform plan`.
+// Workaround: Providing a null-resource for letting Terraform think that there are
+// differences, enabling the Gardener to start an actual `terraform apply` job.
+resource "null_resource" "outputs" {
+  triggers = {
+    recompute = "outputs"
+  }
+}
+
+//=====================================================================
+//= Output Variables
+//=====================================================================
+
+output "router_id" {
```

Comment:
Let's inject the names of the output variables via the Helm values, similar to https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/charts/internal/aws-infra/values.yaml#L27-L36

Reply:
This has not yet been addressed

Reply:
Change to `output "{{ .Values.outputKeys.routerId }}" {`